torch/csrc/distributed/rpc/process_group_agent.cpp

```diff
@@ -542,7 +542,19 @@ void ProcessGroupAgent::listenLoop() {
       recvWork_ = work;
     }
 
-    if (!rpcRunning_.load() || !work->wait() /* aborted */) {
+    if (!rpcRunning_.load()) {
+      return;
+    }
+    bool workCompleted;
+    try {
+      workCompleted = work->wait();
+    } catch (...) {
+      // gloo timed out waiting for the send to succeed, which throws an
```

Comment:
nit: won't this be a receive?

Reply:
The Exception thrown from here is from unbound_buffer.cc in gloo. The error is "Timed out waiting <timeout>ms for send operation to complete"

Reply:
That might be the exception on the client end, but this should throw for recv. Above, we call `pg_->recvAnySource()` which returns a `ProcessGroup::RecvWork`. Tracing down the stack we call `UnboundBuffer::waitRecv` in gloo: https://github.com/facebookincubator/gloo/blob/master/gloo/transport/tcp/unbound_buffer.cc#L54, it looks like. Though I don't think it matters too much, we should just catch whatever exception it throws.